model/GlobalItems.inc.php

```diff
@@ -81,21 +81,48 @@ public static function getGlobalItems($params) {
 		StatsD::timing("api.globalitems", $time * 1000);
 		
 		$code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-		
 		if ($code != 200) {
-			throw new Exception($code . " from Global Items server "
-				. "[requestURL: '$requestURL'] [RESPONSE: '$response']");
+			throw new Exception($code . " from global items server "
+				. "[URL: '$requestURL'] [RESPONSE: '$response']");
+		}
+		
+		$info = curl_getinfo($ch);
+		$headerSize = $info['header_size'];
+		$body = trim(mb_substr($response, $headerSize));
+		$rawHeaders = explode("\n", trim(mb_substr($response, 0, $headerSize)));
+		unset($rawHeaders[0]);
+		$headers = [];
+		foreach ($rawHeaders as $line) {
+			list($key, $val) = explode(':', $line, 2);
+			$headers[strtolower($key)] = trim($val);
 		}
-		return $response;
+		
+		$json = json_decode($body, true);
+		return [
+			'totalResults' => $headers['total-results'],
+			'data' => $json
+		];
 	}
 	
 	public static function getGlobalItemLibraryItems($id) {
 		$params = [
 			'id' => $id
 		];
-		$json = self::getGlobalItems($params);
-		$json = json_decode($json);
-		$libraryItems = $json->libraryItems;
-		return $libraryItems;
+		$result = self::getGlobalItems($params);
```

Comment:
Everything that calls `getGlobalItems` for a single id should account for the possibility that there's no match. For here, if that happened we'd want to return `false` from here and 404 from the controller.